src/app/Command.cpp

```diff
@@ -33,15 +33,14 @@
 namespace chip {
 namespace app {
 
-CHIP_ERROR Command::Init(Messaging::ExchangeManager * apExchangeMgr, InteractionModelDelegate * apDelegate)
+CHIP_ERROR Command::Init(Messaging::ExchangeManager * apExchangeMgr)
```

Comment:
Well, we have split init no matter what, in that the IM delegate is passed to the constructor.

I do question the value the Init method, by the way: the allocations it does could happen when someone tries to actually do something that needs the memory, instead of requiring up-front init...

> I think the pattern in this PR of not permitting re-use (i.e Init can only be called once ever for a given object)

I don't think this PR does that.  It does require a Shutdown and new Init before you can reuse the object.

> If I'm an embedded device that wants to keep re-using the same CommandSender object (which extends this) to send multiple commands in sequence

You can't re-use it until the response to the first command returns, right?  And then you can just destroy and re-create the object, I'd think.  That is, you can allocate memory for the object and then placement-new + delete as needed into that memory, if you want to keep reusing the memory.

So as a followup to this PR I think we should pretty strongly consider requiring that you create this object once, use it, then destroy it.  It makes the lifetime bits very clear.